Services/AwsRoute53Provider.cs

```
@@ -0,0 +1,223 @@
+using Amazon;
+using Amazon.Route53;
+using Amazon.Route53.Model;
+using KCert.Models; // Assuming CertClient is in here for GenerateNewKey, or some other model
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace KCert.Services;
+
+[Service]
+public class AwsRoute53Provider : IDnsProvider
```

General note: I think all the checks for `_route53Client == null` are unnecessary. Let the client fail early if it's misconfigured, and then assume everything is correctly configured otherwise.

The code will be smaller and easier to read.